modules/thunder_updater/thunder_updater.drush.inc

```diff
@@ -34,20 +39,37 @@ function thunder_updater_drush_command() {
 /**
  * Handler for "thunder-updater-generate-update" drush command.
  *
- * @param string $module
- *   Module name.
- * @param string $updateName
- *   Update name.
+ * @param string $patchFilePath
+ *   Filepath where the patch file will be created.
+ * @param string $moduleList
+ *   Comma-seperated.
  *
  * @return array
  *   Returns messages for output.
  */
-function drush_thunder_updater_generate_update($module, $updateName) {
+function drush_thunder_updater_generate_update($patchFilePath, $moduleList = '') {
+
   /** @var \Drupal\thunder_updater\ConfigHandler $configHandler */
   $configHandler = Drupal::service('thunder_updater.config_handler');
 
-  $successful = $configHandler->generatePatchFile($module, $updateName);
-  $message = ($successful) ? dt('Patch file is successfully generated.') : dt('There are no changes that should be exported.');
+  if ($moduleList) {
+    $modules = explode(',', $moduleList);
+  }
+  else {
+    $modules = array_filter(\Drupal::moduleHandler()->getModuleList(), function (Extension $extension) {
+      return ($extension->getType() == 'module');
+    });
+    $modules = array_keys($modules);
+  }
+  if ($regex = drush_get_option('filter', '')) {
+    $modules = array_filter($modules, function ($module) use ($regex) {
+      return preg_match($regex, $module);
+    });
+  }
+
+  $patchFile = $configHandler->generatePatchFile($modules);
+  file_put_contents($patchFilePath, $patchFile, FILE_CREATE_DIRECTORY);
```

Comment:
I think this is not correct, since: FILE_CREATE_DIRECTORY - is provided by Drupal and should not be used with: `file_put_contents`.

Additionally, file should not be created if `$patchFile` is empty.